token-metadata/program/src/utils/fees.rs

```diff
@@ -0,0 +1,86 @@
+use solana_program::{
+    account_info::AccountInfo, entrypoint::ProgramResult, program::invoke, rent::Rent,
+    sysvar::Sysvar,
+};
+
+use crate::{
+    error::MetadataError,
+    state::{
+        fee::{CREATE_FEE, UPDATE_FEE},
+        Key, Metadata, TokenMetadataAccount, MASTER_EDITION_FEE_FLAG_INDEX,
+        METADATA_FEE_FLAG_INDEX,
+    },
+};
+
+#[cfg(feature = "serde-feature")]
+use serde::{Deserialize, Serialize};
+
+#[repr(C)]
+#[cfg_attr(feature = "serde-feature", derive(Serialize, Deserialize))]
+#[derive(PartialEq, Eq, Debug, Clone, Copy)]
+pub enum IxType {
+    CreateMetadata,
+    UpdateMetadata,
+}
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+pub(crate) struct LevyArgs<'a> {
+    pub ix_type: IxType,
+    pub payer_account_info: &'a AccountInfo<'a>,
+    pub token_metadata_pda_info: &'a AccountInfo<'a>,
+}
+
+pub(crate) fn levy(args: LevyArgs) -> ProgramResult {
+    // Fund metadata account with rent + Metaplex fee.
+    let rent = Rent::get()?;
+
+    let fee = match args.ix_type {
+        IxType::CreateMetadata => CREATE_FEE + rent.minimum_balance(Metadata::size()),
```

Comment:
Should there be a check to determine if the rent needs to be included here? Could this add the rent value twice if the account is already initialized?

Reply:
Yeah, you're right for `create`. Fixed.